src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -843,14 +976,67 @@ public Random random()
     }
 
-    public List<CompactionAggregate.UnifiedAggregate> maybeSort(List<CompactionAggregate.UnifiedAggregate> pending)
+    /**
+     * Return the overlap inclusion method to use when combining overlap sections into a bucket. For example, with
+     * SSTables A(0, 5), B(2, 9), C(6, 12), D(10, 12) whose overlap sections calculation returns [AB, BC, CD],
+     *   - NONE means no sections are to be merged. AB, BC and CD will be separate buckets, compactions AB, BC and CD
+     *     will be added separately, thus some SSTables will be partially used / single-source compacted, likely
+     *     to be recompacted again with the next selected bucket.
+     *   - SINGLE means only overlaps of the sstables in the selected bucket will be added. AB+BC will be one bucket,
+     *     and CD will be another (as BC is already used). A middle ground of sorts, should reduce overcompaction but
+     *     still has some.
+     *   - TRANSITIVE means a transitive closure of overlapping sstables will be selected. AB+BC+CD will be in the same
+     *     bucket, selected compactions will apply to all overlapping sstables and no overcompaction will be done, at
+     *     the cost of reduced compaction parallelism and increased length of the operation.
+     * TRANSITIVE is the default and makes most sense. NONE is a closer approximation to operation of legacy UCS.
+     * The option is exposed for experimentation.
+     */
+    public OverlapInclusionMethod overlapInclusionMethod()
+    {
+        return overlapInclusionMethod;
+    }
+
+    public static int[] parseScalingParameters(String str)
+    {
+        String[] vals = str.split(",");
+        int[] ret = new int[vals.length];
+        for (int i = 0; i < vals.length; i++)
+        {
+            String value = vals[i].trim();
+            int W = UnifiedCompactionStrategy.parseScalingParameter(value);
+            ret[i] = W;
+        }
+
+        return ret;
+    }
+
+    public static String printScalingParameters(int[] parameters)
     {
-        return env.maybeSort(pending);
+        StringBuilder builder = new StringBuilder();
+        int i;
+        for (i = 0; i < parameters.length - 1; ++i)
+        {
+            builder.append(UnifiedCompactionStrategy.printScalingParameter(parameters[i]));
+            builder.append(", ");
+        }
+        builder.append(UnifiedCompactionStrategy.printScalingParameter(parameters[i]));
+        return builder.toString();
     }
 
-    public IntArrayList maybeRandomize(IntArrayList aggregateIndexes)
+    /**
+     * Prioritize the given aggregates. Because overlap is the primary measure we aim to control, reducing the max
+     * overlap of the aggregates is the primary goal. We do this by sorting the aggregates by max overlap, so that
+     * the ones with the highest overlap are chosen first.
+     * Among choices with matching overlap, we order randomly to give each level and bucket a good chance to run.
```

Comment:
Because this version of UCS (unlike the OSS one) reserves threads per level and only leaves the remainder to be arbitrarily distributed, it was much less obvious if putting a preference wouldn't break a promise of spreading work (close to) equally among levels.

I'm in the process of changing the per-level thread reservations so this may yet be changed depending on some fallout tests results.